File: ts/src/connections/forwardedPortConnectingEventArgs.ts
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

import { SshStream } from '@microsoft/dev-tunnels-ssh';

/**
 * Forwarded port connecting event args.
 */
export class ForwardedPortConnectingEventArgs {
    /**
     * Creates a new instance of ForwardedPortConnectingEventArgs.
     */
    public constructor(
        /**
         * Gets the forwarded port
         */
        public readonly port: number,

        /**
         * Gets the stream
         */
        public readonly stream: SshStream,
    ) {}
}

File: cs/src/Connections/ForwardedPortConnectingEventArgs.cs
// <copyright file="RefreshingTunnelAccessTokenEventArgs.cs" company="Microsoft">
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license.
// </copyright>

using Microsoft.DevTunnels.Ssh;
using System;

namespace Microsoft.DevTunnels.Connections;

/// <summary>
/// Event args for the forwarded port connecting event.
/// </summary>
public class ForwardedPortConnectingEventArgs : EventArgs {
    /// <summary>
    /// Creates a new instance of <see cref="ForwardedPortConnectingEventArgs"/> class.
    /// </summary>
    public ForwardedPortConnectingEventArgs(uint port, SshStream stream) {
        Port = port;
        Stream = stream;
    }

    /// <summary>
    ///  Forwarded port
    /// </summary>
    public uint Port { get; }

    /// <summary>
    /// Connection stream
    /// </summary>
    public SshStream Stream { get; }
}
